### ISSUE TITLE
Run JAX gradient checks

### ISSUE THREAD

**Issue by eldavenport** [climate-analytics-lab/jax-gcm#172]
JAX has a built-in tool for this. Our unit tests check that we don't get NaNs, but this should provide a more robust test. It could help identify functions that are not differentiable at certain points. It would be nice to incorporate this into unit tests.

https://docs.jax.dev/en/latest/_autosummary/jax.test_util.check_grads.html
